modules/account-lib/src/coin/eos/BuyRamBytesActionBuilder.ts

```diff
@@ -0,0 +1,99 @@
+import * as EosJs from 'eosjs';
+import { TransactionBuilder as EosTxBuilder } from 'eosjs/dist/eosjs-api';
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { EosActionBuilder } from './eosActionBuilder';
+import { Action } from './ifaces';
+import { BuyRamBytesActionSchema } from './txnSchema';
+
+export class BuyRamBytesActionBuilder extends EosActionBuilder {
+  private _payer: string;
+  private _receiver: string;
+  private _bytes: number;
+
+  constructor(act: Action) {
+    super(act);
+    this.action.name = this.actionName();
+  }
+
+  /**
+   * Sets the account name of the payer
+   *
+   * @returns {this} this action builder.
+   *
+   * @param {string} payer valid eos name
+   */
+  payer(payer: string): this {
+    this._payer = payer;
+    return this;
+  }
+
+  /**
+   * Sets the account name of the reciever
+   *
+   * @returns {this} this action builder.
+   *
+   * @param {string} receiver valid eos name
+   */
+  receiver(receiver: string): this {
+    this._receiver = receiver;
+    return this;
+  }
+
+  /**
+   * Sets the bytes quantity
+   *
+   * @returns {this} this action builder.
+   *
+   * @param {number} bytes valid eos quantity
+   */
+  bytes(bytes: number): this {
+    this._bytes = bytes;
+    return this;
+  }
+
+  /**
+   * Get action name
+   *
+   * @returns {string} The name of the action e.g. transfer, buyrambytes, delegatebw etc
+   */
+  actionName(): string {
+    return 'buyrambytes';
+  }
+
+  /** @inheritdoc */
+  build(builder: EosTxBuilder): EosJs.Serialize.Action {
+    const data = this.action.data;
+    if (typeof data === 'string') {
```

Comment:
Why the check for data type as string vs undefined (or not present)?

Reply:
Data is returned as a string when we deserialize the transaction without actions. Deserializing with actions returns the data as an object. Planning to move this check into the parent class in the next PR/Commit